services/imagery/src/image-store.js

```diff
@@ -16,42 +19,93 @@ export default class ImageStore extends EventEmitter {
    * get the folder ready.
    *
-   * Image metadata is stored as JSON. A file called count.json holds
-   * the number of the images taken, so when the image stores starts
+   * Image metadata is stored as JSON. A SQLite database holds
+   * a list of the images taken, so when the image stores starts
    * with an existing directory, it can use the previous images.
    *
    * The rate at which images are added is also stored. Note that
    * this does not take in consideration the timestamp of the images.
+   *
+   * A limit can be placed on the maximum number of images stored by
+   * setting the maxImages parameter. When the limit is reached,
+   * the store will begin deleting old images.
    */
 
   /** Create a new image store. */
-  constructor(clearExisting = false) {
+  constructor(clearExisting = false, maxImages = undefined) {
     super();
 
     this._clearExisting = clearExisting;
-    this._count = 0;
+    this._maxImages = maxImages;
 
     // The time of the last images in the store.
     this._times = [];
+
+    this._dbPool = null;
   }
 
   /** Creates an empty directory for the image store if needed. */
   async setup() {
     if (this._clearExisting === true) {
       await fs.emptyDir(FOLDER_NAME);
-    } else if (await fs.exists(COUNT_FILE)) {
-      // Reading the file and getting the count from there.
-      this._count = JSON.parse(await fs.readFile(COUNT_FILE)).count;
-    } else {
-      // If there isn't a count file, we'll make sure this directory
-      // exists.
-      await fs.mkdirp(FOLDER_NAME);
     }
+
+    // If there isn't a database file, we'll make sure this
+    // directory exists.
+    await fs.mkdirp(FOLDER_NAME);
+
+    // A connection pool prevents database queries from intefering
+    // with each other while a transaction is occurring. Instead,
+    // connections will wait on each others' transactions to finish
+    // before proceeding execution or starting another transaction.
+    this._dbPool = genericPool.createPool({
+      create: () => sqlite.open(DB_FILE),
+      destroy: (db) => db.close()
+    }, { max: 5, min: 0 });
+
+    let db = await this._dbPool.acquire();
+    await db.run('CREATE TABLE IF NOT EXISTS ' +
+      'images(id INTEGER PRIMARY KEY AUTOINCREMENT)');
+    this._dbPool.release(db);
   }
 
   /** Get the number of images stored. */
-  getCount() {
-    return this._count;
+  async getCount() {
+    let db = await this._dbPool.acquire();
+    const count = (await db.get('SELECT COUNT(id) FROM images'))['COUNT(id)'];
+    this._dbPool.release(db);
+
+    return count;
+  }
+
+  /** Get a list of image IDs available for retrieval. */
+  async getAvailable() {
+    let db = await this._dbPool.acquire();
+    const available = (await db.all('SELECT id FROM images ORDER BY id ASC'))
+      .map(row => row.id);
+    this._dbPool.release(db);
+
+    return available;
+  }
+
+  /** Get the ID of the last image stored. */
+  async getLatestId() {
```

Comment:
Can you get a test case in the image-store test file for `getLatestId()` and `exists()`. These are being tested at all since in the router tests they are mocked.